### PR TITLE
Add group_concat support

### LINKED ISSUE #1715: I think h2's group_concat should be supported
The h2 function list http://www.h2database.com/html/functions.html has a group_concat operation.

Could you add support for this feature? How can I add this function myself until there's support on the main release?

Thanks :)

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1715

### CODE REVIEW

**Comment by activedecay** [comment]
when do you think we'll see this groupConcat integrated into a release?

**Comment by timowest** [comment]
next release at the end of january

File: querydsl-sql/src/test/java/com/querydsl/sql/SelectBase.java [hunk 1]
```diff
@@ -2040,5 +2040,33 @@ public void end(SQLListenerContext context) {
         assertTrue(endCalled.get() - getResultsCalled >= 100);
     }
 
+    @Test
+    @ExcludeIn(DERBY)
+    public void groupConcat() {
```
Comment (robertandrewbain) [inline querydsl-sql/src/test/java/com/querydsl/sql/SelectBase.java]: Poorly named

File: querydsl-sql/src/test/java/com/querydsl/sql/SelectBase.java [hunk 2]
```diff
@@ -2040,5 +2040,33 @@ public void end(SQLListenerContext context) {
...
+        assertEquals(
+                expected,
+                query().select(SQLExpressions.groupConcat(employee.firstname))
+                        .from(employee)
+                        .groupBy(employee.superiorId).fetch());
+    }
+
+    @Test
+    @ExcludeIn(DERBY)
+    public void groupConcat2() {
```
Comment (robertandrewbain) [inline querydsl-sql/src/test/java/com/querydsl/sql/SelectBase.java]: Poorly named

**Comment by timowest** [comment]
I'd keep the test names like that, we can use more verbose test names for new test classes, but for query integration tests we have lots of operator named test methods, and I wouldn't change that practice in existing classes.

**Comment by Shredder121** [comment]
Looks good, works fine.

**Comment by johnktims** [comment]
Looks good.

**Pull request merged**